libs/base/Data/Strings.idr

```diff
@@ -16,15 +16,19 @@ partial
 foldl1 : (a -> a -> a) -> List a -> a
 foldl1 f (x::xs) = foldl f x xs
 
--- This works quickly because when string-append builds the result, it allocates
+-- This works quickly because when string-concat builds the result, it allocates
 -- enough room in advance so there's only one allocation, rather than lots!
+%foreign
+    "scheme:string-concat"
+    "node:lambda:__prim_idris2js_stringConcat"
```

Comment:
AFAIK if we want this to work on node and the browser `node` should be replaced with `javascript` according to https://github.com/idris-lang/Idris2/blob/master/docs/source/backends/javascript.rst#javascript-ffi-specifiers

Reply:
Thank you!